File: examples/java/java-booking-03_preserve_whole_object/src/main/java/tv/codely/booking/Tax.java
package tv.codely.booking;

public final class Tax {
    private final TaxType type;
    private final TaxValue value;

    public Tax(TaxType type, TaxValue value) {
        this.type  = type;
        this.value = value;
    }

    public TaxType type() {
        return type;
    }

    public TaxValue value() {
        return value;
    }
}

File: examples/csharp/csharp-booking-04_tell_dont_ask/src/Booking/Tax.cs
namespace CodelyTv.Booking
{
    public sealed class Tax
    {
        private readonly TaxType type;
        private readonly TaxValue value;

        public Tax(TaxType type, TaxValue value)
        {
            this.type = type;
            this.value = value;
        }
    }
}
